src/engine/attributes/set.js

```diff
@@ -1,33 +1,34 @@
 import "../../../src/engine/helper/setMatcher.js";
-import { convertToString } from "../../../src/utils/lang/index.js";
 
 export default function setApply(val, userVal, op){
 
-	let vals = convertToString(val)
-	const ruleSet = new Set();
-	vals.split(",").map((item)=>{
-		ruleSet.add(item.trim())
-	}) 
+	if (userVal === undefined || userVal === null) {
+		throw new Error("Invalid set attribute value. Set attribute must be type of set or array.")
```

Comment:
should this instead say not null and non empty?

Reply:
HOLD for Mahrukh's review @Mahrukhizhar

Reply:
Do you mean we should allow null value in case of Set type attribute? null => empty set, I think we can allow empty set to be passed as its a valid set value

Reply:
yeah we should allow empty set